dex/networks/eth/abi.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2019 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
```

Comment:
I _think_ it's OK to mix **L**GPL.  Fortunately they only use gpl for the actual geth executable app.

Reply:
OK, so feedback from @jrick ... the benefit of lgpl over gpl only applies when _dynamically linking_.  While we are an open source project and it doesn't affect us, any proprietary software building on dcrdex would be affected unless they rip out these bits.  If we revisit the plugins idea, which is is dynamically linked libraries if I understand that correctly, it would be alright, but plugins sucked for a number of reasons including cgo.
Thoughts?

Reply:
As long as we only import go-ethereum from obviously eth-specific packages (we do), a third consumer can fairly easily avoid accidentally importing (and statically linking) LGPL code.

The design we have of only the actual applications like dexc and dcrdex importing the asset drivers is fortunate since all other importable packages in our repository, other than the `.../eth` ones, will be free of the LGPL code.

To make sure this doesn't change, I suggest we comment or make a readme in each package that either imports go-ethereum code or has such lifted code, describing clearly that by importing the package the consumer is linking LGPL code.

Reply:
gate all eth usage behind a build tag.